### DISCUSSION TITLE
Additional Collection Type: Blog

### DISCUSSION THREAD

**Discussion by lorenzolewis** [withastro/starlight#948] [Feature Requests]
### What version of `starlight` are you using?

0.5.5

### What is your idea?

Add a new content collection for blog posts with support for front matter such as published date, author(s), and an adapted layout type.

### Why is this feature necessary?

A lot of competitors in the field have built-in (or easily add-able) blog functionality. You _could_ build on top of the existing `docs` content collection for a blog functionality (and I've seen a few cases on Discord for this), but a blog is a completely different use case that could benefit from it's own collection.

Things that come to mind are:
- Frontmatter for author(s), published date, tag, etc.
- A way to define authors and their metadata
- An adapted layout for blog posts
- A derived layout to list all of the blog posts
- RSS support
- "Share" links to share a post to media platforms

### Do you have examples of this feature in other projects?

- Gatsby: https://www.gatsbyjs.com/starters/gatsbyjs/gatsby-starter-blog/
- Docusaurus: https://docusaurus.io/docs/blog
- VuePress: https://github.com/vuepress/vuepress-plugin-blog
- 11ty: https://github.com/11ty/eleventy-base-blog

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by lorenzolewis** [comment]
@HiDeoo being amazing as always: https://github.com/HiDeoo/starlight-blog

**Comment by HiDeoo** [comment]
Oh, didn't see there was a feature request for this ^^

From what you mentioned, I think only RSS feeds and share links are not supported yet. I basically needed an initial version out there with a basic set of features but I do plan on adding more in the future. I have not research it yet but I think both should be possible and not that much work.

The major downside of the integration is how it is currently implemented and how hacky it is, but this should get better with time and when we get a better vision of how customizations will be handled in Starlight.

Of course, having this built-in could also be really neat and if at any point the team decides to implement it, I'll be happy to archive this integration and help with the implementation if needed ofc.

**Comment by HiDeoo** [comment]
With time and more perspective on this topic since [plugins](https://starlight.astro.build/reference/plugins/) have landed in Starlight and later on [`<StarlightPage>`](https://starlight.astro.build/guides/pages/#using-starlights-design-in-custom-pages), I think it's fair to say that having different type of content handled by different plugins is a totally valid approach. This also avoids the need for Starlight to have a built-in support for every possible use case and reduces the complexity of the core.

Considering all this, I'm going to close this discussion.
